flume-ng-core/src/test/java/org/apache/flume/source/TestMultiportSyslogTCPSource.java

```diff
@@ -383,4 +386,84 @@ public void testPortCharsetHandling() throws UnknownHostException, Exception {
     Assert.assertNull(evt.getHeaders().get(SyslogUtils.EVENT_STATUS));
   }
 
+  /**
+   * Test event with a given ip header.
+   */
+  @Test
+  public void testIpHeader() throws UnknownHostException, IOException {
+    final String TEST_IP_HEADER = "testIpHeader";
+
+    MultiportSyslogTCPSource source = new MultiportSyslogTCPSource();
+    Channel channel = new MemoryChannel();
+
+    Context channelContext = new Context();
+    channelContext.put("capacity", String.valueOf(100));
+    channelContext.put("transactionCapacity", String.valueOf(100));
+    Configurables.configure(channel, channelContext);
+
+    List<Channel> channels = Lists.newArrayList();
+    channels.add(channel);
+
+    ChannelSelector rcs = new ReplicatingChannelSelector();
+    rcs.setChannels(channels);
+
+    source.setChannelProcessor(new ChannelProcessor(rcs));
+    Context context = new Context();
+    StringBuilder ports = new StringBuilder();
+    context.put(SyslogSourceConfigurationConstants.CONFIG_PORTS,
+            String.valueOf(getFreePort()));
+    context.put(SyslogSourceConfigurationConstants.CONFIG_IP_HEADER,
+            TEST_IP_HEADER);
+
+    source.configure(context);
+    source.start();
+
+    //create a socket to send a test event
+    Socket syslogSocket;
+    syslogSocket = new Socket(
+            InetAddress.getLocalHost(), getFreePort());
```

Comment:
This does not work with the current trunk version.
Please rebase the PR onto trunk and fix the test failure.